vertx-db2-client/src/main/java/io/vertx/db2client/impl/drda/ColumnMetaData.java

```diff
@@ -105,21 +107,48 @@ public void setColumnCount(final int numColumns) {
     }
     
     public List<String> getColumnNames() {
+    	List<String> cols = new ArrayList<>(columns_);
+    	for (int i = 0; i < columns_; i++) {
+    		cols.add(i, getColumnName(i));
+    	}
+    	return cols;
+    }
+    
+    public String getColumnName(int i) {
     	// Prefer column names from SQLDXGRP if set
-    	if (!isNull(sqlxName_))
-    		return Collections.unmodifiableList(Stream.of(sqlxName_).collect(Collectors.toList()));
+    	if (sqlxName_ != null && i < sqlxName_.length && sqlxName_[i] != null)
+    		return sqlxName_[i];
     	// Otherwise use column names from SQLDOPTGRP
-    	if (sqlName_ != null && sqlName_.length > 0)
-    		return Collections.unmodifiableList(Stream.of(sqlName_).collect(Collectors.toList()));
-    	return Collections.emptyList();
+    	if (sqlName_ != null && i < sqlName_.length)
+    		return sqlName_[i];
+    	return null;
     }
```

Comment:
Should we add a check to make sure the passed in i >= 0?  Or are we guaranteed a value that is 0+

Reply:
Good catch, I think this method will eventually be an SPI so I'll add the check